plugins/modules/github_gpg_key.py

```diff
@@ -0,0 +1,386 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024, Austin Lucas Lake <53884490+austinlucaslake@users.noreply.github.com>
+# Based on community.general.github_key module by Ansible Project
+
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+module: github_gpg_key
+short_description: Manage GitHub GPG keys
+description:
+  - Creates, removes, or list GitHub GPG keys.
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  token:
+    description:
+      - GitHub OAuth or personal access token (classic) with the read:gpg_key, write:gpg_key, and admin:gpg_key scopes needed to manage GPG keys.
+    required: true
+    type: str
+  name:
+    description:
+      - GPG key name
+    required: true
+    type: str
+  armored_public_key:
+    description:
+      - ASCII-armored GPG public key value. Required when O(state=present).
+    type: str
+  state:
+    description:
+      - Whether to remove a key, ensure that it exists, or update its value.
+    choices: ['present', 'absent']
+    default: 'present'
+    type: str
+  force:
+    description:
+      - The default is V(true), which will replace the existing remote key
+        if it is different than O(pubkey). If V(false), the key will only be
+        set if no key with the given O(name) exists.
+    type: bool
+    default: true
+
+author: Austin Lucas Lake (@austinlucaslake)
+'''
+
+RETURN = '''
+deleted_keys:
+    description: An array of key objects that were deleted. Only present on state=absent
+    type: list
+    returned: When state=absent
+    sample: [{
+        "id": 3,
+        "name": "Octocat's GPG Key",
+        "primary_key_id": 2,
+        "key_id": "3262EFF25BA0D270",
+        "public_key": "xsBNBFayYZ...",
+        "emails": [{
+            "email": "octocat@users.noreply.github.com",
+            "verified": True
+        }],
+        "subkeys": [{
+            "id": 4,
+            "primary_key_id": 3,
+            "key_id": "4A595D4C72EE49C7",
+            "public_key": "zsBNBFayYZ...",
+            "emails": [],
+            "can_sign": False,
+            "can_encrypt_comms": True,
```

Comment:
Generally use `true` and `false` for YAML booleans:
```suggestion
            "can_encrypt_comms": true,
```